artman/cli/converter.py

```diff
@@ -0,0 +1,297 @@
+                if 'lower' in legacy_package_version:
+                    artifact.package_version.grpc_dep_lower_bound = (
+                        legacy_package_version.get('lower'))
+                if 'upper' in legacy_package_version:
+                    artifact.package_version.grpc_dep_upper_bound = (
+                        legacy_package_version.get('upper'))
+
+            # Compute publishing targets
+            if 'git_repos' in legacy_artifact_config:
+                artifact.publish_targets.extend(_compute_publish_targets(
+                    legacy_artifact_config.get('git_repos'), common_git_repos))
+            result.append(artifact)
+    return result
+
+
+def _compute_artifact_type(legacy_common):
+    if legacy_common.get('packaging', '') == 'google-cloud':
+        return Artifact.GAPIC_ONLY
+    elif legacy_common.get('package_type', '') == 'grpc_common':
+        return Artifact.GRPC_COMMON
+
+    return Artifact.GAPIC
+
+
+def _compute_publish_targets(git_repos, common_git_repos):
+    result = []
+    for k, v in git_repos.items():
+        location = (common_git_repos[k]['location']
+                    if k in common_git_repos.keys() else v['location'])
+        target = Artifact.PublishTarget()
+        target.name = k
+        target.location = location
+        target.type = Artifact.PublishTarget.GITHUB
+
+        if 'paths' in v:
+            target.directory_mappings.extend(
+                _compute_directory_mappings(v['paths']))
+        result.append(target)
+    return result
+
+
+def _compute_directory_mappings(paths):
+    result = []
+    for path in paths:
+        mapping = Artifact.PublishTarget.DirectoryMapping()
+        if isinstance(path, str):
+            mapping.dest = path
+        else:
+            if 'src' in path:
+                mapping.src = path.get('src')
+            if 'dest' in path:
+                mapping.dest = path.get('dest')
+            if 'artifact' in path:
+                mapping.name = path.get('artifact')
+            result.append(mapping)
+    return result
+
+
+def _compute_deps(proto_deps):
+    result = []
+    for dep in proto_deps:
+        new_dep = Artifact.ProtoDependency()
+        new_dep.name = dep
+        result.append(new_dep)
+    return result
+
+
+def _compute_src_proto_paths(src_proto_paths):
+    result = []
+    for src_proto_path in src_proto_paths:
+        result.append(_sanitize_repl_var(src_proto_path))
+    return result
+
+
+def _sanitize_repl_var(value):
+    if value.startswith('${GOOGLEAPIS}/'):
+        return value.replace('${GOOGLEAPIS}/', '')
+    if value.startswith('${REPOROOT}/'):
+        return value.replace('${REPOROOT}/', '')
+    return value
+
+def _camel_to_underscore(name):
+    camel_pat = re.compile(r'([A-Z])')
+    return camel_pat.sub(lambda x: '_' + x.group(1).lower(), name)
+
+
+def _convert_json(d):
+    """Convert the dict to turn all key into lower underscore case."""
```

Comment:
Also, not a huge deal, but this is the kind of thing I tend to prefer third-party packages for (e.g. https://pypi.org/project/stringcase/)